### PR TITLE
Enables Support for tracking Orb Releaes in CircleCI Releases

### PR DESCRIPTION
**Pull request by james-crowley**
The changes below add support for CircleCI's Releases. This addition adds an optional parameter to the publish job called `release_environment`. If `release_environment` is set, upon releasing a new version of the orb the release itself will be tracked in the supplied environment. 

Note users will need to make a custom environment type in CircleCI Releases and then pass that environment name as the parameter. 

By default `release_environment` is set to an empty string, which will skip over the release tracking. This should not break any usage of the existing orb tools.

### CODE REVIEW

**Comment by james-crowley** [comment]
@david-montano-circleci This PR is blocked until #236 is merged. The version of the CircleCI is too old to run the release command. 

Which result in an error like:

```
Error: unknown command "run" for "circleci"

Did you mean this?
        runner

Run 'circleci --help' for usage.

Exited with code exit status 255
```

**Comment by james-crowley** [comment]
I updated the PR to avoid cli confusion between `circleci` and `circleci-agent`. The PR no longer requires merging of #236

**Review by david-montano-circleci — APPROVED** [review]

**Pull request merged**
